### PR TITLE
Add Critical/High/Medium/Low for rule and match severity

### PR DESCRIPTION
**Pull request by aryx**
This add supports only in semgrep-core and osemgrep.
Support in pysemgrep (which uses jsonschema for
checking rule correctness) will be in another PR.

test plan:
unit test added
make
make test

### CODE REVIEW

File: src/osemgrep/cli_ci/Ci_subcommand.ml
Comment (aryx) [inline src/osemgrep/cli_ci/Ci_subcommand.ml]: @chmccreery what should be the severity integer value for Critical in the finding type?
(Note that we should really not use an int here ... it would be better to use the match_severity type which is an enum)

File: libs/paths/Ppath.ml
Comment (chmccreery) [inline libs/paths/Ppath.ml:312]: What's up with all these seemingly unrelated changes?

File: src/osemgrep/cli_ci/Ci_subcommand.ml
Comment (chmccreery) [reply src/osemgrep/cli_ci/Ci_subcommand.ml]: I'd expect a 3 - isn't that whatwas in the interfaces PR? is there a reason you jumped to 5? I see that Experiment is 4, but don't see a 3 anywhere.

Also, I've been working on pulling severity directly out of rule_versions rather than storing it with findings anyway, so in the near future hopefully you could do whatever you want with severity (e.g. reuse match_severity type) and the app won't care. But not quite yet...

Comment (chmccreery) [reply src/osemgrep/cli_ci/Ci_subcommand.ml]: nit: can you order these in increasing or descending order of severity for readability?

**Review by chmccreery — COMMENTED** [review]
Someone who writes ocaml should review this as well, but mostly it make sense to me! 

My main concern is that Experiment seems to be handled inconsistently (sometimes it raises an error like in Sarif_output.ml but other times it seems to map to "Unknown" like in Gitlab_output.ml or "Information" like in Convert_utils. Is that intentional/could you explain the pattern? I'm assuming this is because different tools do or don't have an "unknown" type category, but it seems like Experiment rules shouldn't ever really be user-facing anyway - aren't those for internal use only?

**Comment by aryx** [comment]
Yes I'll try to make a separate PR to cleanup this Experiment/Inventory thing.
@p4p3r are we still use those rule severity somewhere?

File: tests/rules/severity_critical.yaml
Comment (emjin) [inline tests/rules/severity_critical.yaml:7]: Should we add a test for each level so that we get a failure if the mappings fail?

**Review by emjin — APPROVED** [review]
Looks fine to me; I don't know much about the severities change so I can't speak to that. Approving to unblock but I expect you/Clara/SR will resolve what the values should be first

**Pull request merged**
